pkg/api/pod/util.go

```diff
@@ -46,11 +46,7 @@ const AllContainers ContainerType = (InitContainers | Containers | EphemeralCont
 // AllFeatureEnabledContainers returns a ContainerType mask which includes all container
 // types except for the ones guarded by feature gate.
 func AllFeatureEnabledContainers() ContainerType {
-	containerType := AllContainers
-	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
-		containerType &= ^EphemeralContainers
-	}
-	return containerType
+	return AllContainers
```

Comment:
Do we want to keep this function now?

Is this a function that was added back then by ephemeralContainers or is it one of these abstractions that we always had "just in case" and we want to keep it?

Reply:
We added the container visitor during ephemeral containers because it tipped complexity of visiting all containers over a threshold. We can retire `AllFeatureEnabledContainers()` if we don't think we're ever going to add another type of container.

If we do it, I recommend decoupling from this feature graduation and do it early in the next dev cycle so we don't cause a bunch of conflicts for people racing the the merge window.

Reply:
SGTM :)

Reply:
I'd probably keep the method, just in case, but agree that either way, it shouldn't be changed in this PR